metaflow/plugins/cards/card_modules/components.py

```diff
@@ -11,6 +12,12 @@
 from .convert_to_native_type import TaskToDict, _full_classname
 from .renderer_tools import render_safely
 
+if TYPE_CHECKING:
+    from pandas import DataFrame
```

Comment:
doesn't this introduce a hard dependency to `PIL`, `pandas`, `matplotlib` which is not desirable?

Reply:
I think it adds a dependency only for type resolution. 
```
>>> import PIL
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'PIL'
>>> from metaflow.plugins.cards.card_modules.components import Table
>>>
```